Makefile

```diff
@@ -123,8 +123,24 @@ license: $(MYGOBIN)/addlicense
 check-license: $(MYGOBIN)/addlicense
 	./hack/add-license.sh check
 
+## lint-api-static runs the linter on the API module 
+## with build-tag kustomize_disable_go_plugin_support
+## this aims to catch any issues with the API module
+## that would prevent the API module from being used
+## when the go plugin support is disabled.
+export KUSTOMIZE_ROOT ?= $(shell pwd | sed -E 's|(.*\/kustomize)/(.*)|\1|')
+.PHONY: lint-api-static
+lint-api-static: $(MYGOBIN)/golangci-lint $(MYGOBIN)/goimports $(builtinplugins)
+	cd api && \
+	$(MYGOBIN)/golangci-lint \
+	  -c $$KUSTOMIZE_ROOT/.golangci.yml \
+	  --build-tags kustomize_disable_go_plugin_support \
+	  -D exhaustive \
+	  --path-prefix api \
+	  run ./...
+
```

Comment:
Since this is specific to `api`, it should go in `api/Makefile` instead.

Reply:
Moved in [33dc9be](https://github.com/kubernetes-sigs/kustomize/pull/5651/commits/33dc9be3ed6c83ccb4450c9dac9ebb17aa59c362)